core/ignore.js

```diff
@@ -132,23 +132,23 @@ class Ignore {
 
   // Add some rules for things that should be always ignored (temporary
   // files, thumbnails db, trash, etc.)
-  addDefaultRules () {
+  addDefaultRules () /*: this */ {
     const defaultPatterns = buildPatternArray(readLinesSync(defaultRulesPath))
     this.patterns = defaultPatterns.concat(this.patterns)
     return this
   }
 
   // Return true if the given file/folder path should be ignored
-  isIgnored (doc /*: Metadata */) {
+  isIgnored (pathLike /*: string */, kind /*: EventKind */) /*: boolean */ {
```

Comment:
What about adding a separate `Ignore` method for this?

Reply:
You'd rather have one method for `Metadata` and one for `AtomWatcherEvent`?

Reply:
I just want to make sure we keep a `Metadata` specific one, since we're still using it with docs in ~10 places.
It could eventually be renamed, e.g. to something like `#isIgnoredDoc()`.

Reply:
Actually `Ignore` only need to know whether something is a folder (aka *boolean*):

```
Ignore#match({path /*: string */, isFolder /*: boolean */})
```

Both `Metadata` and `AtomWatcherEvent` versions could be plain functions using `Ignore#match()`:

```
metadata.shouldIgnore(doc /*: Metadata */, ignoreRules /*: Ignore */)
shouldIgnore(event /*: AtomWatcherEvent */, ignoreRules /*: Ignore */) // defined in ignore step since it is only used there
```

toutes deux utilisant :